Sell/Basket/Repository/BasketRepository.swift

```diff
@@ -0,0 +1,183 @@
+            timeslot: input.timeslotId,
+            fulfilmentPoint: input.fulfilmentPointId,
+            seatInfo: input.seatInfo?.compactMap {
+                ApolloType.SeatInfoInput(key: $0.key, value: $0.value)
+            },
+            items: input.items?.compactMap {
+                ApolloType.BasketItemInput(
+                    product: $0.productId,
+                    productVariant: $0.productVariantId,
+                    fulfilmentPoint: $0.fulfilmentPointId,
+                    quantity: $0.quantity,
+                    productModifierItems: $0.productModifierItems?.compactMap {
+                        ApolloType.ProductModifierItemSelectionInput(
+                            productModifierItemId: $0.productModifierItemId,
+                            quantity: $0.quantity)
+                    }
+                )
+            }
+        )
+    }
+
+    private func makeApolloCheckoutInput(_ input: CheckoutInput) -> ApolloType.CheckoutInput {
+        ApolloType.CheckoutInput(
+            netAmount: input.netAmount,
+            language: ApolloType.Language(rawValue: input.language ?? ""))
+    }
+}
+
+extension BasketRepository: BasketProvidable {
+
+    public func getMyBasket(
+        callback: @escaping (Result<Basket, BasketError>) -> Void
+    ) {
+        graphQLManager.dispatch(
+            query: ApolloType.GetMyBasketQuery(),
+            cachePolicy: .fetchIgnoringCacheCompletely) { result in
+            switch result {
+            case .success(let response):
+                if let transformedError = self.transformErrorIfNeccessary(response.errors) {
+                    callback(.failure(transformedError))
+                } else if let returnedBasket = response.data?.getMyBasket?.fragments.fragmentBasket {
+                    callback(.success(Basket(response: returnedBasket)))
+                } else {
+                    callback(.failure(.emptyBasket))
+                }
+            case .failure(let error):
+                callback(.failure(.regularErrors([error])))
+            }
+        }
+    }
+
+    public func createMyBasket(
+        input: BasketInput,
+        callback: @escaping (Result<Basket, BasketError>) -> Void
+    ) {
+        graphQLManager.dispatchMutation(
+            mutation: ApolloType.CreateMyBasketMutation(input: makeApolloBasketInput(input)),
+            cacheResultToPersistence: false
+        ) { result in
+            switch result {
+            case .success(let response):
+                if let transformedError = self.transformErrorIfNeccessary(response.errors) {
+                    callback(.failure(transformedError))
+                } else if let returnedBasket = response.data?.createMyBasket?.fragments.fragmentBasket {
+                    callback(.success(Basket(response: returnedBasket)))
+                } else {
+                    callback(.failure(.emptyBasket))
```

Comment:
See above comment about failure to unwrap the data not being the same as empty basket.